### PR TITLE
feat: branchPrefixOld

### LINKED ISSUE #15496: branchPrefixOld
### What would you like Renovate to be able to do?

Allow users to change/migrate branch prefix without worrying about recreating old/ignored PRs.

Renovate uses branch name + PR title to determine if it's created an update PR before, and if the old one has been closed by the user then it's treated as ignored and not recreated. Therefore if a user changes branchPrefix (e.g. from `renovate/` to `renovate-`) then any old `renovate/*` PRs will transition from ignored to being recreated.

### If you have any ideas on how this should be implemented, please tell us here.

Support a new config option `branchPrefixOld`, and check it too. Essentially: create new branches with `branchPrefix` but allow checking of existing branches against `branchPrefix` and `branchPrefixOld`.

### Is this a feature you are interested in implementing yourself?

No

### PR DESCRIPTION
**Pull request by Gabriel-Ladzaretti**
## Changes
Ignore closed PRs with `branchName` = `branchPrefixOld/update` in addition to default `branchPrefix/update` ones.

## Context
closes #15496

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by viceice** [comment]
Does thios close #15496? Then please add `- closes` before issue id to autoclose issue after merge

File: lib/workers/repository/update/branch/check-existing.spec.ts
Comment (viceice) [inline lib/workers/repository/update/branch/check-existing.spec.ts:45]: use `partial<Pr>()` from `test/utils` instead of never

File: lib/workers/repository/update/branch/check-existing.spec.ts
Comment (viceice) [inline lib/workers/repository/update/branch/check-existing.spec.ts:49]: same here

**Review by viceice — COMMENTED** [review]
otherwise lgtm

**Review by HonkingGoose — CHANGES_REQUESTED** [review]

**Review by viceice — COMMENTED** [review]
LGTM, waiting for docs approval

**Review by rarkins — COMMENTED** [review]
Have you considered if you can also use this for currently open PRs?

**Comment by Gabriel-Ladzaretti** [comment]
> Have you considered if you can also use this for currently open PRs?

i have not, in this case we will end up with a duplicate pr.
https://github.com/ladzaretti/cfg-testing-json5/pulls

the obvious solution would be to check for an open pr with the `oldprefix` and if there is one, then we need not create a new one.  

WDYT?

**Comment by rarkins** [comment]
> the obvious solution would be to check for an open pr with the `oldprefix` and if there is one, then we need not create a new one.

Yes, that would be ideal. Can you see how complex any implementation would be?

**Comment by Gabriel-Ladzaretti** [comment]
> Yes, that would be ideal. Can you see how complex any implementation would be?

pretty straight forward really, in `branch/index` we first check if `config.branchName` (with branch prefix) exists. if it does not exist, we will check for the same branch but with the old prefix, if this one exists we just need to update `config.branchName` to that and proceed as is.

i have this working and tested all scenarios i could think of, and it works as it should.

just need to add coverage thats it. all in all its pretty much the same logic we have already implemented in `check-existing` the only diff is that here we are searching for `open` prs thats it.

File: docs/usage/configuration-options.md
Comment (HonkingGoose) [inline docs/usage/configuration-options.md:316]: Is the text still correct now that you made code changes?

Comment (Gabriel-Ladzaretti) [reply docs/usage/configuration-options.md:316]: you are right, we do need to add something about an existing open PR with `branchPrefixOld`.
that said, what we already have is still relevant though

File: docs/usage/configuration-options.md
Comment (Gabriel-Ladzaretti) [inline docs/usage/configuration-options.md:320]: This might need a change as well, seems off in this context
WDYT ?

File: docs/usage/configuration-options.md
Comment (Gabriel-Ladzaretti) [inline docs/usage/configuration-options.md:322]: line 321 might need a change as well, seems off in this context
WDYT ?

Comment (HonkingGoose) [reply docs/usage/configuration-options.md:322]: I'm not happy with the new texts. The sentences don't feel properly linked together conceptually. So the whole section needs to be looked at and fixed. 😉 

I'll let the maintainers check the technical accuracy first, before I make suggestions that don't match the code. 🙈

Comment (viceice) [reply docs/usage/configuration-options.md:322]: i think this is technically right, but can be more detailed.

**Review by viceice — COMMENTED** [review]
code LGTM

File: docs/usage/configuration-options.md
Comment (rarkins) [inline docs/usage/configuration-options.md:322]:
```suggestion
Renovate uses branch names as part of its checks to see if an update PR was created previously, and already merged or ignored.
If you change `branchPrefix`, then no previously closed PRs will match, which could lead to Renovate recreating PRs in such cases.
Instead, set the old `branchPrefix` value as `branchPrefixOld` to allow Renovate to look for those branches too, and avoid this happening.
```

Comment (rarkins) [reply docs/usage/configuration-options.md:322]: @HonkingGoose wdyt?

File: lib/config/options/index.ts
Comment (rarkins) [inline lib/config/options/index.ts:1222]: This is a clever idea to automatically solve the problem for the majority of users who would only change the branchPrefix once?

File: lib/workers/repository/update/branch/check-existing.ts
Comment (rarkins) [inline lib/workers/repository/update/branch/check-existing.ts:31]: I would like a log message here if we found a matching PR with branchPrefixOld

File: lib/workers/repository/update/branch/index.ts
Comment (rarkins) [inline lib/workers/repository/update/branch/index.ts:95]: debug log a message here too

**Review by rarkins — CHANGES_REQUESTED** [review]

File: docs/usage/configuration-options.md
Comment (HonkingGoose) [reply docs/usage/configuration-options.md:322]: I like it!

File: lib/config/options/index.ts
Comment (Gabriel-Ladzaretti) [reply lib/config/options/index.ts:1222]: dont know about clever, it just seemed logical. As this flag is basically just a placeholder for the previous value of `branchPrefix`.

**Review by rarkins — APPROVED** [review]

**Review by viceice — APPROVED** [review]

**Pull request merged**
